Docs: Add 12-factor how-to sections for integrations (#2219)
canonical/charmcraft: docs/howto/manage-web-app-charms/integrate-web-app-charm.rst

- ``POSTGRESQL_DB_HOSTNAME``
- ``POSTGRESQL_DB_PORT``

Integrate with ingress
----------------------

If you wish to integrate your 12-factor web app with an ingress,
for instance
`Nginx Ingress Integrator <https://charmhub.io/nginx-ingress-integrator>`_,

> provide the integration to your deployed 12-factor app with:

Reviewer: ```suggestion
provide the integration to your deployed app with:
```

.. code-block:: bash

    juju integrate <app charm> nginx-ingress-integrator

You do not need to add an endpoint definition to your web app charm's
project file.

Integrate with observability
----------------------------

If you wish to integrate your 12-factor web app with the `Canonical
Observability Stack
(COS) <https://charmhub.io/topics/canonical-observability-stack>`_,
you need to set up an ingress for COS and provide the IP address
of your machine to MicroK8s through the MetalLB addon:

.. code-block:: bash

  IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
  microk8s enable metallb:$IPADDR-$IPADDR

Deploy and integrate observability to the 12-factor app with:

.. code-block:: bash

  juju deploy cos-lite --trust
  juju integrate <app charm> grafana
  juju integrate <app charm> prometheus

>   juju integrate <app charm> loki

Reviewer: For consistency, the indentation should be 4 spaces.

> You do not need to add endpoint definitions to your web app charm's

Reviewer: ```suggestion
You don't need to add endpoint definitions to your charm's
```

If you wish to integrate your 12-factor web app with the `Canonical
Observability Stack
(COS) <https://charmhub.io/topics/canonical-observability-stack>`_,
you need to set up an ingress for COS and provide the IP address

> of your machine to MicroK8s through the MetalLB addon:

Reviewer 2: Perhaps change this to say that COS relies on Traefik which requires the metallb loadbalancer to be enabled on microk8s which requires an IP range

Author: I gave it a rewrite, let me know if you have any other feedback! Thanks!

If you wish to integrate your 12-factor web app with the `Canonical
Observability Stack
(COS) <https://charmhub.io/topics/canonical-observability-stack>`_,
you must first prepare an ingress for COS as it relies on Traefik.
Traefik requires the MetalLB Loadbalancer addon on MicroK8s which requires

> an IP range. Provide the IP range and enable the addon with:

Reviewer 2: How about:

```
COS relies on the Traefik ingress to expose, for example, Grafana. On MicroK8s, Traefik requires the MetalLB loadbalancer to be enabled which requires an IP range. Provide the IP range and enable the addon with:
```

Author: Thanks for the suggestion -- incorporated! I rephrased the first sentence a bit because it sounded off to me -- but I think the explanation about ingress and why we're enabling the addon is much clearer now :)
